src/hue.js

```diff
@@ -0,0 +1,24 @@
+import make from "./make";
+
+export const setHue = (color, hueValue) => {
+  if (hueValue === null) return color;
+  const { s, l, a } = color;
+  return make.fromHsl(hueValue, s, l, a);
+}
+
+export const spin = (color, degrees = 0) => {
+  const { h } = color;
+  const newHue = h + degrees;
+
+  if (newHue < 0) {
+    return setHue(color, 360 - Math.abs(newHue));
+  }
+
+  if (newHue > 360) {
+    return setHue(color, newHue - 360);
+  }
+
+  return setHue(color, newHue);
```

Comment:
```
return setHue(color, Math.abs(newHue) % 360);
```

and you no longer need those other if branches. You wouldn't have to worry about numbers < -360 either.